.prettierrc.json

```diff
@@ -0,0 +1,28 @@
+{
+  "singleQuote": true,
+  "tabWidth": 2,
+  "useTabs": false,
+  "printWidth": 80,
+  "endOfLine": "auto",
```

Comment:
The default Prettier style is used as standard in Arduino tooling projects. Instead of setting a non-standard global Prettier configuration, I recommend applying this custom configuration only to the specific file types for which you don't want to use the default Prettier style.

Reply:
Can you please explain? I do not understand what's wrong—the liners you have marked [were already in IDE2](https://github.com/arduino/arduino-ide/pull/2199/files/ed42b343a87d54994336d811ed87b41b29fa8c44#diff-663ade211b3a1552162de21c4031fcd16be99407aae5ceecbb491a2efc43d5d2L2-L6).

They're on the `main`:

https://github.com/arduino/arduino-ide/blob/f0704b678cbc844faa5738d6f884a98a021d96b2/.prettierrc#L2-L6

Reply:
Yes, but you are now formalizing that Prettier formatting be done to all files in the repository, whereas previously the project infrastructure only applied them to the JavaScript/TypeScript files. That meant that implicitly this custom configuration was only applicable to those file types.

Reply:
> Yes, but you are now formalizing that Prettier formatting be done to all files in the repository,

Apologies, I cannot follow you. Can you please show me where exactly this change is? The formatter config was in `.prettierrc`. I moved the config to `prettierrc.json`. So far my change is NOOP. Then, I added custom rules to JSON, HTML, and CSS file types because I wanted them to have the same style all over the repo. What's the breaking change here? Thanks

Reply:
The change is the `format` script added by this PR here:

https://github.com/arduino/arduino-ide/blob/ed42b343a87d54994336d811ed87b41b29fa8c44/package.json#L56

The `.` in that command causes Prettier to format all files in the repository.

Reply:
I see. I can remove it. It's not run by anything workflow or action.

Reply:
I removed the `format` npm script.